src/SprykerEco/Yves/Payone/Handler/ExpressCheckout/QuoteHydrator.php

```diff
@@ -178,20 +190,25 @@ protected function hydrateQuoteWithAddresses(
         QuoteTransfer $quoteTransfer,
         PayonePaypalExpressCheckoutGenericPaymentResponseTransfer $details
     ) {
+        $address = new AddressTransfer();
+
+        if ($this->customerClient->isLoggedIn()) {
+            $address->setEmail($quoteTransfer->getCustomer()->getEmail());
+        } else {
+            $address->setEmail($details->getEmail());
+        }
 
-        $shippingAddress = new AddressTransfer();
-        $shippingAddress->setFirstName($details->getShippingFirstName());
-        $shippingAddress->setLastName($details->getShippingLastName());
-        $shippingAddress->setCompany($details->getShippingCompany());
-        $shippingAddress->setEmail($details->getEmail());
-        $shippingAddress->setAddress1($details->getShippingStreet());
-        $shippingAddress->setAddress2($details->getShippingAddressAdition());
-        $shippingAddress->setCity($details->getShippingCity());
-        $shippingAddress->setState($details->getShippingState());
-        $shippingAddress->setIso2Code($details->getShippingCountry());
-        $shippingAddress->setZipCode($details->getShippingZip());
-        $quoteTransfer->setBillingAddress($shippingAddress);
-        $quoteTransfer->setShippingAddress($shippingAddress);
+        $address->setFirstName($details->getShippingFirstName());
+        $address->setLastName($details->getShippingLastName());
+        $address->setCompany($details->getShippingCompany());
+        $address->setAddress1($details->getShippingStreet());
+        $address->setAddress2($details->getShippingAddressAdition());
+        $address->setCity($details->getShippingCity());
+        $address->setState($details->getShippingState());
+        $address->setIso2Code($details->getShippingCountry());
+        $address->setZipCode($details->getShippingZip());
+        $quoteTransfer->setBillingAddress($address);
```

Comment:
as far as I remember, in Paypal it's possible to have different shipping & billing addresses.

Reply:
When I was placing order there was only ship to address at least during express checkout.

Reply:
that's strange and true.
If you don't have requirements from PayPal about implementation, then keep it like this.